udTest/src/udChunkedArrayTests.cpp

```diff
@@ -182,7 +185,111 @@ TEST(udChunkedArrayTests, Iterator)
     EXPECT_EQ(i, x);
     ++i;
   }
-  EXPECT_EQ(15, i);
+  EXPECT_EQ(array.length, i - 1);
+
+  udChunkedArrayIterator<int> it = array.begin();
+  //++
+  ++it;
+  EXPECT_EQ(*it, 2);
+
+  //--
+  --it;
+  EXPECT_EQ(*it, 1);
+
+  //+=
+  it += 3;
+  EXPECT_EQ(*it, 4);
+
+  //-=
+  it -= 2;
+  EXPECT_EQ(*it, 2);
+
+  //[]
+  EXPECT_EQ(it[3], 5);
+  EXPECT_EQ(*it, 2); //unchanged
+
+  //defintion of comparisons
+  EXPECT_EQ(it < it, false);
+  EXPECT_EQ(it > it, false);
+  EXPECT_EQ(it <= it, true);
+  EXPECT_EQ(it >= it, true);
+
+  EXPECT_EQ(it != it, false);
+  EXPECT_EQ(it == it, true);
+
+  it = it + 8;
+
+  EXPECT_EQ(it < it + 3, true);
+  EXPECT_EQ(it < it - 1, false);
+
+  EXPECT_EQ(it > it + 3, false);
+  EXPECT_EQ(it > it - 1, true);
+
+  auto it2 = it - 1;
+  EXPECT_EQ(it2 < it, true);
+  EXPECT_EQ(it2 > it, false);
+  EXPECT_EQ(it2, --it);
+
+  it2 = it - 2;
+  --it;
+  EXPECT_EQ(it2, --it);
+
+  it = it + 2;
+  EXPECT_EQ(it - 2, it2);
+  EXPECT_EQ(it > it2, true);
+  EXPECT_EQ(it < it2, false);
+  ++it2;
+  ++it2;
+  EXPECT_EQ(it, it2);
+
+  i = 1;
+  for (auto iter = array.begin(); iter < array.end(); ++iter)
+  {
+    EXPECT_EQ(*iter, i);
+    auto iter2 = array.begin() + i;
+    EXPECT_EQ(*(iter2 - 1), *iter);
+    ++i;
+  }
+  EXPECT_EQ(i, array.length + 1);
+
+  for (auto iter = array.end() - 1; iter > array.begin(); --iter)
+  {
+    --i;
+    EXPECT_EQ(*iter, i);
+    EXPECT_EQ(array.begin()[i - 1], *iter);
+  }
+
+  EXPECT_EQ(array.end() - array.begin(), array.length);
+
+  array.Deinit();
+  array.Init(16);
+  for (int i = 0; i < 500; ++i) // ensure that the array is forced to grow
+  {
+    array.PushBack(i % 4);
+  }
+
+  //non zero inset:
+  array.PopFront();
+
+  std::sort(array.begin(), array.end());
+  int previous = array[0];
+  for (int el : array)
+  {
+    EXPECT_EQ(previous > el, false);
```

Comment:
```suggestion
    EXPECT_TRUE(previous <= el);
```

Testing the positive case makes more sense for something like this.